Assets/Scripts/CastButtonController.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CastButtonController : MonoBehaviour
+{
+    bool isActive;
+    SpriteRenderer spriteRenderer;
+    public Sprite redSprite;
+    public Sprite greySprite;
+
+    public int cardsRequired;
+
+    void Start() {
+        spriteRenderer = GetComponent<SpriteRenderer>();
+    }
+
+    public void OnSelectionChanged(int numSelected) {
+        bool oldActive = isActive;
+
+        isActive = numSelected == cardsRequired ? true : false;
+
+        if (isActive != oldActive) {
+            spriteRenderer.sprite = isActive ? redSprite : greySprite;
+        }
```

Comment:
Kind of confusing logic here, would maybe rewrite to something like:
```suggestion
        
     isActive = numSelected == cardsRequired;
     spriteRenderer.sprite = isActive ? redSprite : greySprite;
     
```

Reply:
The isActive line should definitely be changed to that lol.

The reason I had the if statement is so that the sprite wouldn't be changing unnecessarily. (It only changes if the state of the button changes.)

Do you think this slight optimization is worth the confusing code?

Reply:
I wouldn't say so, it took me some time to figure out the gist of what was going on haha. Also we're doing an extra comparison call so the optimization is debatable?